.github/workflows/lint.yml

```diff
@@ -0,0 +1,50 @@
+name: Lint
+
+on:
+  workflow_call: {}
+
+permissions:
+  contents: read
+
+jobs:
+  lint-rust:
+    name: Rust Lint
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Rust
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          components: rustfmt, clippy
+
+      - name: Rust Fotmat
+        run: cargo fmt --all -- --check
+
+      - name: Clippy
+        run: cargo clippy --all --all-features -- -D warnings
+
+  lint-commits:
+    name: Commit Lint
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+  
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+
+      - name: Install commitlint
+        run: npm install --save-dev @commitlint/{cli,config-conventional}
+
+      - name: Commitlint (PR)
+        if: github.event_name == 'pull_request'
+        run: npx commitlint --from ${{ github.event.pull_request.base.sha }}
+    
+      - name: Commitlint (Push)
+        if: github.event_name == 'push'
+        run: npx commitlint --from ${{ github.event.before }}
```

Comment:
Based on my limited reading, sometimes `github.event.before` is empty? Like the first commit of a branch?

Reply:
I also think that releases and versioning are essential as soon as someone depends on this package. Already I could start using what you've written to get started on other projects. So might as well start releasing. The version number is not for marketing or for aesthetics - so there's no harm on being on v201.0.0 as far as I'm concerned!

TL;DR releases are useful immediately so we might as well start making them.

Reply:
I also think you'll be surprised at how little your API will change. The interface you created won't change that much with the refactoring we have planned. Introducing new stuff is not a breaking change. So I think it'll be fine.

Reply:
Hmm, yeah fair enough. If you can already start using this for other projects might as well release versions so we can keep track of things.

I will read more into this commitlint to understand how it works.